cpp/ql/lib/ext/std.vector.model.yml

```diff
@@ -0,0 +1,59 @@
+extensions:
+  - addsTo:
+      pack: codeql/cpp-all
+      extensible: summaryModel
+    data: # namespace, type, subtypes, name, signature, ext, input, output, kind, provenance
+      - ["std", "vector", True, "assign", "", "", "Argument[*@1]", "Argument[-1].Element[@]", "value", "manual"]
+      - ["std", "vector", True, "assign<InputIt>", "(InputIt,InputIt)", "", "Argument[0]", "Argument[-1].Element[@]", "value", "manual"]
+      - ["std", "vector", True, "at", "", "", "Argument[-1].Element[@]", "ReturnValue[*@]", "value", "manual"]
+      - ["std", "vector", True, "back", "", "", "Argument[-1].Element[@]", "ReturnValue[*@]", "value", "manual"]
+      - ["std", "vector", True, "begin", "", "", "Argument[-1].Element[@]", "ReturnValue.Element[@]", "value", "manual"]
+      - ["std", "vector", True, "cbegin", "", "", "Argument[-1].Element[@]", "ReturnValue.Element[@]", "value", "manual"]
+      - ["std", "vector", True, "data", "", "", "Argument[-1].Element[@]", "ReturnValue[*@]", "value", "manual"]
+      - ["std", "vector", True, "emplace", "", "", "Argument[*@1]", "Argument[-1].Element[@]", "value", "manual"]
+      - ["std", "vector", True, "emplace", "", "", "Argument[*@1]", "ReturnValue[*@]", "value", "manual"]
```

Comment:
Should the `emplace` models flow to `ReturnValue.Element[@]`, since the return value is an iterator?

Reply:
Well spotted! Fixed in 6d053247244c9bc3e4db727451a0edc96a076001. There are some fun discrepancies where some `emplace`-like functions return [references](https://en.cppreference.com/w/cpp/container/vector/emplace_back), and others return [iterators](https://en.cppreference.com/w/cpp/container/vector/emplace). I _think_ I've gotten them right now 🤞